### PR TITLE
🎨 [Frontend] Improve Notification texts and Bell's UX

### PR DESCRIPTION
**Pull request by odeimaiz**
## What do these changes do?

This PR adds two more fields to the Notifications: ``resource_id`` and ``user_from_id``.

Thanks to these new fields, the Notification UI element now contains more meaningful information:
```
{Resource} {NameOfResource}
was shared by {nameOfSharer}
{DateAndTime}
```

Also when tapping the bell, all the notifications will be marked as read.

New UI:
image

Marked as Read:
MarkAllAsRead

## Related issue/s

- closes https://github.com/ITISFoundation/osparc-issues/issues/934
- closes https://github.com/ITISFoundation/osparc-issues/issues/1011
- closes https://github.com/ITISFoundation/osparc-issues/issues/1770
- closes https://github.com/ITISFoundation/osparc-simcore/issues/4459

## Dev-ops checklist

- [ ] No ENV changes or I properly updated ENV ([read the instruction](https://git.speag.com/oSparc/osparc-ops-deployment-configuration/-/blob/configs/README.md?ref_type=heads#how-to-update-env-variables))

### CODE REVIEW

**Review by ignapas — APPROVED** [review]

File: services/web/server/src/simcore_service_webserver/users/_notifications.py [hunk 1]
```diff
@@ -26,8 +26,10 @@ class NotificationCategory(StrAutoEnum):
 
 class BaseUserNotification(BaseModel):
     user_id: UserID
+    user_from_id: Literal[None] | UserID = None
     category: NotificationCategory
     actionable_path: str
+    resource_id: Literal[""] | str = ""
     title: str
```
Comment (sanderegg) [inline services/web/server/src/simcore_service_webserver/users/_notifications.py:33]: are these not always filled up?

**Review by sanderegg — APPROVED** [review]
nice!

File: services/web/server/src/simcore_service_webserver/users/_notifications.py (hunk 1 above)
Comment (odeimaiz) [reply services/web/server/src/simcore_service_webserver/users/_notifications.py:33]: Nope, not in already existing notifications. The new ones will contain that information, but I still need to be able to read the old ones.

I want to I default them to those values. Is that correct?

File: services/web/server/src/simcore_service_webserver/users/_notifications.py [hunk 2]
```diff
@@ -106,6 +108,8 @@ class Config:
                     "date": "2023-02-23T16:28:13.122Z",
                     "product": "s4l",
                     "read": False,
+                    "resource_id": "3fb96d89-ff5d-4d27-b5aa-d20d46e20e12",
+                    "user_from_id": "2",
```
Comment (GitHK) [inline services/web/server/src/simcore_service_webserver/users/_notifications.py:112]: Q: what about existing notifications which do not have this field set. Can we have an example that is still validated against one such model?

Comment (odeimaiz) [reply services/web/server/src/simcore_service_webserver/users/_notifications.py:112]: You previous 3 examples do not have these fields 10 lines above...

**Review by jsaq007 — APPROVED** [review]
👍🏼

**Review by matusdrobuliak66 — APPROVED** [review]

**Pull request merged**